### PR TITLE
feat: offline login persistence

### LINKED ISSUE #233: Handle the sign-in persistence edge-cases
Currently the signin screen uses a systematic call that fails if the device is offline, thus voiding any offline feature of the app. I see two ways of fixing this issue:
- use the [FirebaseAuth.AuthStateListener](https://firebase.google.com/docs/reference/android/com/google/firebase/auth/FirebaseAuth.AuthStateListener) class to get a callback when the app needs to handle one of the documented edge-cases
- handle gracefully the failure while being offline. In other words, instead of simply failing the operation and holding the user in the signin screen we should differentiate between a failed operation due to a lack of connectivity or due to an invalid token

I didn't thought about the second option initially but it seems way simpler to implement and, to the best of my knowledge, shouldn't leave any loose end

### PR DESCRIPTION
**Pull request by loggerz**
Closes #233

### CODE REVIEW

File: app/src/main/java/com/github/se/travelpouch/ui/authentication/SignIn.kt [hunk 1]
```diff
@@ -196,14 +205,19 @@ fun SignInScreen(
...
-                  .value) { // extra condition on sign-in completion to avoid unnecessary
+                  .value && isConnectedToNetwork.value != null) { // extra condition on sign-in completion to avoid unnecessary
             // recompositions
             LaunchedEffect(Unit) {
               try {
                 isLoading.value = true
                 methodChosen.value = true
 
-                currentUser.getIdToken(true).await() // We shouldn't continue until this passes
+                if (isConnectedToNetwork.value == true) {
```
Comment (ArthurChalard) [inline app/src/main/java/com/github/se/travelpouch/ui/authentication/SignIn.kt:215]: Are you sure that you have to test the something is equat to true ?

**Review by ArthurChalard — CHANGES_REQUESTED** [review]
Nice catch. I almost forgot that we had to manage this case. The helper is also a good idea as it separates the code

File: app/src/main/java/com/github/se/travelpouch/ui/authentication/SignIn.kt (hunk 1 above)
Comment (loggerz) [reply app/src/main/java/com/github/se/travelpouch/ui/authentication/SignIn.kt:215]: Yes I am, keep in mind it's not a Boolean but a Boolean? where the null state represents the helper in the initial state (very short but still a case to handle)

File: app/src/main/java/com/github/se/travelpouch/model/authentication/NetworkConnectivityHelper.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.util.Log
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+class NetworkConnectivityHelper(private val context: Context) {
```
[the remark this answers is not in the document]
Comment (loggerz) [reply app/src/main/java/com/github/se/travelpouch/model/authentication/NetworkConnectivityHelper.kt:12]: Sure!

**Review by ArthurChalard — APPROVED** [review]
Perfect, the change is perfect for me

**Pull request merged**